### ISSUE TITLE
Docker-compose example config doesn't seem to work

### ISSUE THREAD

**Issue by amilbourne** [localstack/localstack#1177]
In the top-level `README.md`, under the 'Running in Docker' section, you give an example docker-compose config file.  Perhaps I'm doing something wrong but this doesn't seem to work.  I'm using docker compose v 1.23.2 and the environment variable substitution generates errors of the form:

    ERROR: Invalid interpolation format for "ports" option in service "localstack": "${PORT_WEB_UI-8080}:${PORT_WEB_UI-8080}"

It seems that while the standard environment variable syntax `${NAME}` is supported, the extended syntax `${NAME-default}` is not.  This doesn't seem to have made it into the docker compose docs on master, but has been documented here:
https://github.com/docker/compose/blob/8cc8e61/docs/compose-file.md
(go to the end of the document)
This documentation could be out of date, but removing the default clauses did indeed fix the problem, so it seems to be accurate.

Have I missed something, or is the example wrong?

**Comment by joarleymoraes** [comment]
For a working `docker-compose` configuration, you should add the `version`. Here's an example:

```
version: '3'
services:
  localstack:
    image: localstack/localstack
    ports:
      - "4567-4584:4567-4584"
      - "${PORT_WEB_UI-8080}:${PORT_WEB_UI-8080}"
    environment:
      - SERVICES=${SERVICES- }
      - DEBUG=${DEBUG- }
      - DATA_DIR=${DATA_DIR- }
      - PORT_WEB_UI=${PORT_WEB_UI- }
      - LAMBDA_EXECUTOR=${LAMBDA_EXECUTOR- }
      - KINESIS_ERROR_PROBABILITY=${KINESIS_ERROR_PROBABILITY- }
      - DOCKER_HOST=unix:///var/run/docker.sock
    volumes:
      - "${TMPDIR:-/tmp/localstack}:/tmp/localstack"

```

**Comment by whummer** [comment]
Hi @amilbourne , I'm unable to reproduce this issue.
```
$ docker-compose -v
docker-compose version 1.23.2, build 1110ad01
$ TMPDIR=/private$TMPDIR docker-compose up
Removing localstack_localstack_1
Recreating 191b1c9cc505_localstack_localstack_1 ... done
Attaching to localstack_localstack_1
localstack_1  | Waiting for all LocalStack services to be ready
localstack_1  | 2019-03-09 17:38:18,638 INFO Set uid to user 0 succeeded
localstack_1  | 2019-03-09 17:38:18,651 INFO supervisord started with pid 6
...
```

Does it work if you specify the `version` attribute, as mentioned by @joarleymoraes above?

**Comment by ManfredLange** [comment]
As of writing I'm using docker-compose version 1.23.2 (docker version 18.09.2) and the following docker-compose.yml file works for me:
```
version: '3.7'

services:
  localstack:
    image: localstack/localstack
    ports:
      - "4567-4593:4567-4593"
      - "${PORT_WEB_UI-8080}:${PORT_WEB_UI-8080}"
    environment:
      - SERVICES=${SERVICES- }
      - DEBUG=${DEBUG- }
      - DATA_DIR=${DATA_DIR- }
      - PORT_WEB_UI=${PORT_WEB_UI- }
      - LAMBDA_EXECUTOR=${LAMBDA_EXECUTOR- }
      - KINESIS_ERROR_PROBABILITY=${KINESIS_ERROR_PROBABILITY- }
      - DOCKER_HOST=unix:///var/run/docker.sock
    volumes:
      - "${TMPDIR:-/tmp/localstack}:/tmp/localstack"
      - "/var/run/docker.sock:/var/run/docker.sock"
```
**Note:** port ranges are now up to 4593 as additional endpoints are now available.

**Comment by amilbourne** [comment]
This all makes sense and I can indeed make it run with version 2.1 and 3).  But on line 119 of `README.md` it says: 'currently supports docker-compose version 2'.  Hence I used version 2, but I now see that 2.1 is specified in the `docker-compose.yaml` that is checked in (I should have thought to look at this).  If version 2.1 is needed then the readme should probably be updated.  I created PR #1183 to fix this.
